addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -941,7 +941,7 @@ export function isUnremovable(node) {
         isEditableRoot ||
         (node.nodeType === Node.ELEMENT_NODE &&
             (node.getAttribute('t-set') || node.getAttribute('t-call'))) ||
-        (node.classList && node.classList.contains('oe_unremovable'))
+        (node.matches && node.matches('.oe_unremovable, .o_submit, [data-toggle="tab"], [data-toggle="collapse"]'))
     );
 }
 
```

Comment:
Note sure if this is the best way to do it, this could lead to a lot of bloat in a utility function if we end up wanting to prevent the deletion of different nodes. Maybe in the snippet editor context we should be able to blacklist elements so that their nodes can't be deleted by Odoo editor.